src/deluge/gui/views/instrument_clip_view.cpp

```diff
@@ -5622,28 +5694,32 @@ void InstrumentClipView::rotateNoteRowHorizontally(ModelStackWithNoteRow* modelS
 
 		UI* currentUI = getCurrentUI();
 
-		// Always shift automation when in Automation View
-		// or also shift automation when default setting to only shift automation in Automation View is false
-		bool shiftAutomation = (currentUI == &automationView || !FlashStorage::automationShift);
+		// If you're in Automation View, only shift automation if you're not in the Note Editor
+		// or also shift Automation when default setting to only shift automation in Automation View is false
+		bool shiftAutomation = ((currentUI == &automationView && !automationView.inNoteEditor())
+		                        || (currentUI != &automationView && !FlashStorage::automationShift));
 
-		// Always clear Notes and MPE when you're not in Automation View
-		bool shiftSequenceAndMPE = (currentUI != &automationView);
+		// If you're in Automation View, only shift Notes and MPE if you're in the Note Editor
+		// Always shift Notes and MPE when you're not in Automation View
+		bool shiftSequenceAndMPE =
+		    ((currentUI != &automationView) || (currentUI == &automationView && automationView.inNoteEditor()));
```

Comment:
there's enough differences here that I almost feel like note automation should be its own view?

Reply:
Lots of checks for whether you're in note editor or not in this PR

Reply:
Will split out the views in the next PR.